src/modules/solvers/QrispQIRO.py

```diff
@@ -0,0 +1,198 @@
+                "values": [2, 3, 4, 5],
+                "description": "How QIRO reps (Parameter: n) do you want? Choose this parameter sensibly in relation to"
+                               " the graph size!"
+            }
+        }
+
+    class Config(TypedDict):
+        """
+        Attributes of a valid config.
+
+        .. code-block:: python
+
+            shots: int
+            depth: int
+            iterations: int
+            QIRO_reps: int
+
+        """
+        shots: int
+        depth: int
+        iterations: int
+        QIRO_reps: int
+
+    def run(self, mapped_problem: any, device_wrapper: any, config: Config, **kwargs: dict) -> (any, float, dict):
+        """
+        Run Qrisp QIRO on the local Qrisp simulator
+
+        :param mapped_problem: Dictionary with the keys 'graph' and 't'
+        :param device_wrapper: Instance of device
+        :param config:
+        :param kwargs: no additionally settings needed, may include the measurement kwargs
+        :return: Solution, the time it took to compute it and optional additional information
+        """
+
+        n = config["QIRO_reps"]
+        p = config["depth"]
+        shots = config["shots"]
+        max_iter = config["iterations"]
+
+        g = mapped_problem['graph']
+        start = start_time_measurement()
+        
+        qiro_instance = QIROProblem(g,
+                                    replacement_routine=create_maxIndep_replacement_routine,
+                                    cost_operator=create_maxIndep_cost_operator_reduced,
+                                    mixer=qiro_RXMixer,
+                                    cl_cost_function=maxIndepSetclCostfct,
+                                    init_function=qiro_init_function
+                                    )
+        qarg = QuantumVariable(g.number_of_nodes())
+        # Run actual optimization algorithm
+        try:
+            res_qiro = qiro_instance.run_qiro(qarg=qarg, depth=p, n_recursions=n, mes_kwargs={"shots": shots},
+                                              max_iter=max_iter)
+        except ValueError as e:
+            logging.error(f"The following ValueError occurred in module QrispQIRO: {e}")
+            logging.error("The benchmarking run terminates with exception.")
+            raise Exception("Please refer to the logged error message.") from e
+        best_bitstring = self._qiro_select_best_state(res_qiro, maxIndepSetclCostfct(g))
+
+        def _translate_state_to_nodes(state: str, nodes: list) -> list:
+            return [key for index, key in enumerate(nodes) if state[index] == '1']
+        
+        winner_state = _translate_state_to_nodes(best_bitstring, nodes=g.nodes())
+
+        return winner_state, end_time_measurement(start), {}
+
+    # post_processing: find the best solution out of the 10 most likely ones.
+    # QIRO is an optimization algo, the most_likely solution is mostly of bad quality, unless the problem is complex
```

Comment:
I think these comments should be part of the docstring of _qiro_select_best_state

Reply:
done!